### PR TITLE
Blog App: Controllers specs

### PR DESCRIPTION
**Pull request by forLoop94**
### The following changes were made:

- I added RSpec as a gem to the project.
- I created a Request spec file for all controllers.
- I made sure that for each action I checked:

1.   If response status was correct.
2.   If a correct template was rendered.
3.   If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    it 'Should get all posts by user with id of 7' do
+      get '/users/7/posts'
+      expect(response).to have_rendered('index')
+      expect(response.body).to include 'Pagination'
+      expect(response).to have_http_status(:success)
+    end
+    it 'Should get the post with id of 1 by user with id of 7' do
+      get '/users/7/posts/1'
+      expect(response).to have_rendered('show')
+      expect(response.body).to include 'Post #1 by username'
```
Comment (iambenkis) [inline spec/requests/posts_spec.rb]: - [ ] Good job creating controllers specs 👏 . I've noticed that you have used a matcher provided by the Capybara testing framework, however as for this project you are supposed to use `rspec-rails` gem `render_template`. Kindly update that . Note that both methods are used to confirm that the expected views are being rendered in a Ruby on Rails application.

```ruby
expect(response).to render_template(:show)
```

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      expect(response.body).to include 'User Name'
+      expect(response).to have_rendered('index')
+    end
+    it 'Should get details of the user with id of 7' do
+      get '/users/7'
+      expect(response).to have_http_status(:success)
+      expect(response.body).to include 'See all posts'
+      expect(response).to have_rendered('show')
+    end
+  end
```
Comment (iambenkis) [inline spec/requests/users_spec.rb]: - [ ] As for the previous comments , Kindly use `render_template` instead of  ` have_rendered`

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi @forLoop94  

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

Good Job adding tests for all controllers ✔️ 
✔️ All tests passed 
✔️ Work documented professionally

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by forLoop94** [comment]
Thank you @iambenkis, for your recommendation, it has been implemented.

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @forLoop94       , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  . 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] TESTS are passing
- [x] Professional Readme




 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**